gitlab-push/handler.go

```diff
@@ -0,0 +1,212 @@
+package function
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/openfaas/openfaas-cloud/sdk"
+)
+
+const Source = "gitlab-push"
+
+// Handle a serverless request
+func Handle(req []byte) string {
+
+	event := os.Getenv("Http_X_Gitlab_Event")
+
+	if event != "Push Hook" && event != "System Hook" {
+		auditEvent := sdk.AuditEvent{
+			Message: "bad event: " + event,
+			Source:  Source,
+		}
+
+		sdk.PostAudit(auditEvent)
+
+		return fmt.Sprintf("%s cannot handle event: %s", Source, event)
+	}
+
+	xGitlabToken := os.Getenv("Http_X_Gitlab_Token")
+
+	if readBool("validate_token") {
+		tokenSecretKey, secretErr := sdk.ReadSecret("gitlab-webhook-secret")
+		if secretErr != nil {
+			return secretErr.Error()
+		}
+
+		if matchErr := tokenMatch(xGitlabToken, tokenSecretKey); matchErr != nil {
+			return matchErr.Error()
+		}
+	}
+
+	gitlabPushEvent := sdk.GitLabPushEvent{}
+	err := json.Unmarshal(req, &gitlabPushEvent)
+	if err != nil {
+		return err.Error()
+	}
+
+	pushEvent := sdk.PushEvent{
+		Ref: gitlabPushEvent.Ref,
+		Repository: sdk.Repository{
+			Name:     gitlabPushEvent.GitLabProject.Name,
+			FullName: gitlabPushEvent.GitLabProject.PathWithNamespace,
+			CloneURL: gitlabPushEvent.GitLabRepository.CloneURL,
+			Owner: sdk.Owner{
+				Login: gitlabPushEvent.GitLabProject.Namespace,
+				Email: gitlabPushEvent.UserEmail,
+			},
+		},
+		AfterCommitID: gitlabPushEvent.AfterCommitID,
+		Installation: sdk.Installation{
+			ID: gitlabPushEvent.GitLabProject.ID,
+		},
+	}
+
+	var found bool
+
+	if readBool("validate_customers") {
+
+		customersURL := os.Getenv("customers_url")
+
+		customers, getErr := getCustomers(customersURL)
```

Comment:
I found a bug in the code in the original version, can you sync up please?

Reply:
Should I add it as separate commit to see the difference or amend to this one?